### PR TITLE
Fix/table commands 2.0

### LINKED ISSUE #82: Error in addTable invalid command
Add table command error doesn't tell the user that a tag is required for tableID.

Image

### LINKED ISSUE #120: Create test cases for AddTableCommandParser
Currently, there are no AddTableCommandParser, leading to errors in the displayed text

### LINKED ISSUE #124: Perhaps refactor tableId/ to be tid/ -> similar to how guestId is implemented to make it for users.

### PR DESCRIPTION
**Pull request by bhavinaa**
This PR updates and improves the implementation of table-related commands to ensure better error handling, input validation, and user experience. The following changes have been made:

### What's Changed
- Refactored `AddTableCommand` to include proper checks for:
  - Duplicate table IDs
  - Invalid table IDs or capacity values (negative or zero)
  - Ensuring a wedding is currently active before adding a table
- Improved error messages to be more specific and user-friendly
- Enhanced `DeleteTableCommand` to handle bad inputs more gracefully
- Added `GetAllTablesCommand` to display tables with proper formatting
- Strengthened input validation in `AddTableCommandParser` and `DeleteTableCommandParser`
- Added comprehensive unit tests for `AddTableCommand`, `DeleteTableCommand`, and their parsers

### Testing
- Manual testing of all table-related commands (`addTable`, `deleteTable`, `getTables`)
- JUnit tests updated and extended to cover edge cases and validation errors

### Notes
- Merge conflicts with `master` were resolved
- Synced changes with `upstream/master` to ensure compatibility
Fixes #82 
Fixes #120 
Fixes #124

### CODE REVIEW

**Comment by bhavinaa** [comment]
it does not build! throws an error right now!

File: src/main/java/seedu/address/logic/commands/AddTableCommand.java [hunk 1]
```diff
@@ -36,21 +35,43 @@ public AddTableCommand(int tableId, int capacity) {
...
+    //
+    //     } catch (TableNotFoundException tnfe) {
+    //
+    //         Table table = new Table(tableId, capacity);
+    //         model.addTable(table);
+    //
+    //         return new CommandResult(String.format(MESSAGE_SUCCESS, tableId, capacity));
+    //     }
+    //
+    // }
```
Comment (josejhkim) [inline src/main/java/seedu/address/logic/commands/AddTableCommand.java:54]: If these are comments I think they can be removed or just reduced to 1-2 lines as a 'old logic'

File: src/main/java/seedu/address/logic/parser/AddTableCommandParser.java [hunk 2]
```diff
@@ -24,12 +24,30 @@ public AddTableCommand parse(String args) throws ParseException {
...
+        String[] tokens = args.trim().split("\\s+");
+        if (tokens.length != 2) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddTableCommand.MESSAGE_USAGE));
+        }
+
+        try {
+            int tableId = Integer.parseInt(tokens[0]);
+            int capacity = Integer.parseInt(tokens[1]);
+            return new AddTableCommand(tableId, capacity);
+        } catch (NumberFormatException e) {
```
Comment (josejhkim) [inline src/main/java/seedu/address/logic/parser/AddTableCommandParser.java:45]: I think this is a nice idea but it could introduce unwanted bugs so it might be better to keep the command formatting as it is

File: src/main/java/seedu/address/logic/parser/DeleteTableCommandParser.java [hunk 3]
```diff
@@ -10,25 +9,53 @@
...
         ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_TABLE_ID);
 
-        if (!argMultimap.getValue(PREFIX_TABLE_ID).isPresent()) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteTableCommand.MESSAGE_USAGE));
+        // ✅ Handle prefixed format: deleteTable tableId/2
+        if (argMultimap.getValue(PREFIX_TABLE_ID).isPresent()) {
+            return parseWithPrefix(argMultimap);
+        }
+
+        // ✅ Handle raw format: deleteTable 2
```
Comment (josejhkim) [inline src/main/java/seedu/address/logic/parser/DeleteTableCommandParser.java:26]: Same concern as addTable - migth be better to stick with only 1 command formatting

**Review by josejhkim — APPROVED** [review]
Overall looks good to me!
